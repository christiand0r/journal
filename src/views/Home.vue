<template>
  <div
    class="flex flex-col items-center justify-center gap-4 h-screen bg-zinc-900 text-white"
  >
    <h1 class="font-black text-4xl">Bienvido a Journal</h1>
    <div class="flex items-center justify-center gap-4">
      <button
        @click="enterJournal"
        class="font-semibold rounded-lg px-6 py-2 bg-green-700 hover:bg-green-800"
      >
        Entrar
      </button>
      <a href="#TODOAgregarGithubXD" class="font-semibold">Github</a>
    </div>
  </div>
</template>

<script>
export default {
  name: "Home",

  methods: {
    enterJournal() {
      this.$router.push({ name: "noEntry" });
    },
  },
};
</script>

<style></style>
