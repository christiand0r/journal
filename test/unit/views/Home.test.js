import { shallowMount } from "@vue/test-utils";

// Components
import Home from "@/views/Home.vue";
import { expect } from "vitest";

describe("Test in the Home View", () => {
  test("Make a snapshot to Home", () => {
    const wrapper = shallowMount(Home);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Click on button enter should redirect to "no-entry"', () => {
    const mockRouter = {
      push: vi.fn(),
    };

    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "noEntry" });
  });
});
