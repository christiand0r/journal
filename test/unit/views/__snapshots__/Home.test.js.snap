// Vitest Snapshot v1

exports[`Test in the Home View > Make a snapshot to Home 1`] = `
"<div class=\\"flex flex-col items-center justify-center gap-4 h-screen bg-zinc-900 text-white\\">
  <h1 class=\\"font-black text-4xl\\">Bienvido a Journal</h1>
  <div class=\\"flex items-center justify-center gap-4\\"><button class=\\"font-semibold rounded-lg px-6 py-2 bg-green-700 hover:bg-green-800\\"> Entrar </button><a href=\\"#TODOAgregarGithubXD\\" class=\\"font-semibold\\">Github</a></div>
</div>"
`;
